README.md
<!--
This README describes the package. If you publish this package to pub.dev,
this README's contents appear on the landing page for your package.

For information about how to write a good package README, see the guide for
[writing package pages](https://dart.dev/guides/libraries/writing-package-pages).

For general information about developing packages, see the Dart guide for
[creating packages](https://dart.dev/guides/libraries/create-library-packages)
and the Flutter guide for
[developing packages and plugins](https://flutter.dev/developing-packages).
-->

## Getting Started

Callify OTP Verification Package. Simple and easiest way to verify your users through Miss Call, Voice OTP, SMS OTP and WhatsApp OTP.

## Features

Miss Call OTP Verification.

## Getting started

First, add `callify_kit` as a dependency in your pubspec.yaml file.
```yaml
dependencies:
  callify_kit: ^[latest version]
```

Don't forget to `flutter pub get`.

Then add permission in AndroidManifest.xml
```xml
    <uses-permission android:name="android.permission.READ_CALL_LOG" />
```

## Usage

For Callify OTP - Call The  ```callifyVerification```  Method and pass parameters
- ApiKey
- CampaingID
- Phone Number



```dart
await CallifyService().callifyVerification(apiKey, campaingID, phoneNumber,
callReceivedCallBack: (response){
// in response you will get incoming call number, just need to verify it
});
```

For Miss Call OTP - Call The  ```missCallVerification```  Method and pass parameters
- ApiKey
- CampaingID
- Phone Number



```dart
await CallifyService().missCallVerification(apiKey, campaingID, phoneNumber,
callReceivedCallBack: (response){
// If you are using this package in iOS then you will get incoming call number in response and in Android this package will detect the incoming call and validate it with response number
});
```

For Voice OTP - Call The  ```voiceOtpVerification```  Method and pass parameters
- ApiKey
- CampaingID
- Phone Number



```dart
await CallifyService().voiceOtpVerification(apiKey, campaingID, phoneNumber,
callReceivedCallBack: (response){
// in response you get OTP or Failed
});
```

For SMS OTP - Call The  ```smsOtpVerification```  Method and pass parameters
- User Name
- Password
- Originator
- Phone Number




```dart
await CallifyService().smsOtpVerification(userName, password, originator, phoneNumber,
callReceivedCallBack: (response){
// in response you get OTP or Failed
});
```

For WhatsApp OTP - Call The  ```whatsappOtpVerification```  Method and pass parameters
- ApiKey
- Template ID
- Phone Number




```dart
await CallifyService().whatsappOtpVerification(apiKey, templateId, phoneNumber,
callReceivedCallBack: (response){
// in response you get OTP or Failed 
});
```


CHANGELOG.md
## 0.0.1

* Library for Miss Call OTP Service.

## 0.0.2
* Enums Added on callBack.

## 0.0.3
* Issue Fixed.

## 0.0.4
* Voice OTP Added.
* SMS OTP Added
* WhatsApp OTP Added

## 0.0.5
* Callify OTP Added.

## 0.0.6
* Fixes.


test/callify_kit_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:callify_kit/callify_kit.dart';

void main() {
  test('adds one to input values', () {
    // final calculator = Calculator();
    // expect(calculator.addOne(2), 3);
    // expect(calculator.addOne(-7), -6);
    // expect(calculator.addOne(0), 1);
  });
}


example/test/widget_test.dart
// This is a basic Flutter widget test.
//
// To perform an interaction with a widget in your test, use the WidgetTester
// utility in the flutter_test package. For example, you can send tap and scroll
// gestures. You can also use WidgetTester to find child widgets in the widget
// tree, read text, and verify that the values of widget properties are correct.

import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';

import 'package:example/main.dart';
import 'package:get/get.dart';

void main() {
  testWidgets('Counter increments smoke test', (WidgetTester tester) async {
    // Build our app and trigger a frame.
    await tester.pumpWidget(const GetMaterialApp());

    // Verify that our counter starts at 0.
    expect(find.text('0'), findsOneWidget);
    expect(find.text('1'), findsNothing);

    // Tap the '+' icon and trigger a frame.
    await tester.tap(find.byIcon(Icons.add));
    await tester.pump();

    // Verify that our counter has incremented.
    expect(find.text('0'), findsNothing);
    expect(find.text('1'), findsOneWidget);
  });
}


lib/callify_kit.dart
library callify_kit;

import 'dart:async';
import 'dart:convert';
import 'dart:io';
import 'dart:math';
import 'package:flutter/cupertino.dart';
import 'package:permission_handler/permission_handler.dart';
import 'package:http/http.dart' as http;
import 'package:phone_state/phone_state.dart';

enum Responses { success, failed, permissionError }

class CallifyService {
  var isRunning = false;
  var isVerified = false;
  dynamic callReceivedCallBack;

  static final _random = Random();
  static const maxNumericDigits = 17;

  static int getInteger(int digitCount) {
    if (digitCount > maxNumericDigits || digitCount < 1) {
      throw RangeError.range(0, 1, maxNumericDigits, "Digit Count");
    }
    var digit = _random.nextInt(9) + 1;
    int n = digit;

    for (var i = 0; i < digitCount - 1; i++) {
      digit = _random.nextInt(10);
      n *= 10;
      n += digit;
    }
    return n;
  }

  Future missCallVerification(
      {@required apiKey,
      @required campaingID,
      @required userNumber,
      callReceivedCallBack}) async {
    await [
      Permission.phone,
    ].request();
    var response = await http.get(Uri.parse(
        'https://api.smsits.com/misscall?Apikey=$apiKey&recipient=$userNumber&campid=$campaingID&uniqueId=123456'));

    if (response.statusCode == 200) {
      var jsonResponse = json.decode(response.body);
      if (Platform.isAndroid) {
        callReceivedCallBack(
            check(jsonResponse['Sender'], callReceivedCallBack));
      } else {
        callReceivedCallBack(jsonResponse['Sender']);
      }
    } else {
      callReceivedCallBack(Responses.failed);
    }


  }


  Future callifyVerification(
      {@required apiKey,
        @required campaingID,
        @required userNumber,
        callReceivedCallBack}) async {

    var response = await http.get(Uri.parse(
        'https://api.smsits.com/misscall?Apikey=$apiKey&recipient=$userNumber&campid=$campaingID&uniqueId=123456'));
    if (response.statusCode == 200) {
      var jsonResponse = json.decode(response.body);

      callReceivedCallBack(jsonResponse['Sender']);

    } else {
      callReceivedCallBack(Responses.failed);
    }
  }


  voiceOtpVerification(
      {@required apiKey,
      @required campaingID,
      @required userNumber,
      callReceivedCallBack}) async {
    var otp = getInteger(4);
    var response = await http.get(Uri.parse(
        'https://api.smsits.com/misscall?Apikey=$apiKey&recipient=$userNumber&campid=$campaingID&uniqueId=123456&param1=$otp'));
    if (response.statusCode == 200) {
      callReceivedCallBack(otp);
    } else {
      callReceivedCallBack(Responses.failed);
    }
  }

  smsOtpVerification(
      {@required userName,
        @required password,
        @required originator,
        @required userNumber,
        callReceivedCallBack}) async {

    var otp = getInteger(4);

    var phone = userNumber.toString().substring(1,userNumber.toString().length);
    var response = await http.get(Uri.parse(
        'https://api.smsits.com/Gotp?action=sendmessage&username=$userName&&password=$password&recipient=92$phone&originator=$originator&otpcode=$otp&otptype=1'));
    if (response.statusCode == 200) {
      callReceivedCallBack(otp);
    } else {
      callReceivedCallBack(Responses.failed);
    }
  }

  whatsappOtpVerification(
      {@required apiKey,
        @required templateId,
        @required userNumber,
        callReceivedCallBack}) async {
    var phone = userNumber.toString().substring(1,userNumber.toString().length);

    var otp = getInteger(4);
    var response = await http.get(Uri.parse(
        'https://otac.golive.com.pk/api/whatsapp/otp?apikey=$apiKey&recipient=+92$phone&templateId=$templateId&otac=$otp'));
    if (response.statusCode == 200) {
      var js = json.decode(response.body);
      if(js['response']['status'] == 'success'){
        callReceivedCallBack(otp);
      }
      else{
        callReceivedCallBack(Responses.failed);

      }
    } else {
      callReceivedCallBack(Responses.failed);
    }
  }

  Future<StreamSubscription<PhoneState>> check(number, callBack) async {
    return PhoneState.stream.listen((event) {
      var minus = number.toString().length;

      if (number == event.number?.substring(event.number!.length - minus) &&
          event.status == PhoneStateStatus.CALL_ENDED) {
        callBack(Responses.success);
      }
    });
  }
}


example/lib/controller.dart
import 'dart:io';

import 'package:callify_kit/callify_kit.dart';
import 'package:example/SuccessScreen.dart';
import 'package:example/enums.dart';
import 'package:example/otp_verification.dart';
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';
import 'package:flutter_spinkit/flutter_spinkit.dart';
import 'package:get/get.dart';

class AppController extends GetxController{
  TextEditingController numberController = TextEditingController();
  var isLoad  = false;

  var type;
  var otp;
  var missCallNumber;
  var data = Get.arguments;
  var userOTP;
  @override
  void onInit() {
    // TODO: implement onInit
    super.onInit();

    if(data != null){
      type = data[0];
      update();
    }
  }
  final spinkit =
  SpinKitFoldingCube(itemBuilder: (BuildContext context, int index) {
    return DecoratedBox(
      decoration: BoxDecoration(
        color: index.isEven ? Colors.yellow : Colors.black12,
      ),
    );
  });

  sendRequest(number) {
    print(type);
    isLoad = true;
    update();
    if (type == VerificationType.call) {
      CallifyService().voiceOtpVerification(
          apiKey: '', campaingID: '', userNumber: number,
          callReceivedCallBack: (response) {
            print(response);
            otp = response;
            isLoad = false;
            update();
            if(otp.toString().length == 4){
              Get.to(()=>OtpVerification(param: otp,));
            }
          });
    }
    else if(type == VerificationType.missCall){
      CallifyService().missCallVerification(
          apiKey: '', campaingID: '', userNumber: number,
          callReceivedCallBack: (response) async {
            print(response);
            isLoad = false;
            update();

            if(Platform.isIOS){
              missCallNumber = response.toString();
              update();
                Get.to(()=>OtpVerification(param: response.toString()));

            }
            else {
              if(response == Responses.success){
                Get.to(()=>SuccessScreen());
              }
              else{

                Get.snackbar('Verification', 'Verification Failed, Please Try Again');
              }
            }

          });
    }
    else if(type == VerificationType.callify){
      CallifyService().callifyVerification(
          apiKey: '', campaingID: '', userNumber: number,
          callReceivedCallBack: (response) async {
            print(response);
            isLoad = false;
            update();


              missCallNumber = response.toString();
              update();
              Get.to(()=>OtpVerification(param: response.toString()));



          });
    }
    else if(type == VerificationType.sms){
      CallifyService().smsOtpVerification(userName: '', password: '', originator: '', userNumber: number,
      callReceivedCallBack: (response){
        print(response);
        otp = response;
        isLoad = false;
        update();
        if(otp.toString().length == 4){
          Get.to(()=>OtpVerification(param: otp,));
        }
      });
    }
    else if(type == VerificationType.whatsapp){
      CallifyService().whatsappOtpVerification(apiKey: '', templateId: '', userNumber: number,
      callReceivedCallBack: (response){
        print(response);
        otp = response;
        isLoad = false;
        update();
        print(otp);
        if(otp.toString().length == 4){
          Get.to(()=>OtpVerification(param: otp,));
        }
      });
    }
  }

 mobileNumberCheck(code){
  userOTP = code;
  update();
 }
 otpCodeCheck(otp){
    userOTP = otp;
    update();
 }
}

example/lib/phone_screen.dart
import 'package:example/controller.dart';
import 'package:flutter/material.dart';
import 'package:flutter_pin_code_fields/flutter_pin_code_fields.dart';
import 'package:get/get.dart';

class PhoneScreen extends StatelessWidget {
  const PhoneScreen({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return GetBuilder<AppController>(
        init: AppController(),
        builder: (app)=>Scaffold(
      appBar: AppBar(
        backgroundColor: Colors.yellow,
        centerTitle: true,
        title: const Text('Callify',style: TextStyle(color: Colors.black),),
      automaticallyImplyLeading: true,),
      body: Center(

        child: ListView(

          padding: EdgeInsets.all(15.0),
          children: <Widget>[
            const SizedBox(height: 20,),


            TextField(
              controller: app.numberController,
              keyboardType: TextInputType.number,
              decoration: InputDecoration(
                contentPadding: EdgeInsets.all(20),
                labelText: 'Enter Your Mobile Number',
                labelStyle: TextStyle(color: Colors.black54),
                counterText: '',
                hintText: '',
                border: OutlineInputBorder(
                  borderRadius: BorderRadius.circular(30.0),
                  borderSide:
                  const BorderSide(color: Colors.black, width: 2.0),
                ),
                focusedBorder: OutlineInputBorder(
                  borderRadius: BorderRadius.circular(30.0),
                  borderSide:
                  const BorderSide(color: Colors.black, width: 2.0),
                ),
              ),

            ),
            SizedBox(height: 20,),
            app.isLoad ? app.spinkit : Container(

                height: MediaQuery.of(context).size.height * 0.07,


                decoration: BoxDecoration(
                    borderRadius: BorderRadius.circular(MediaQuery.of(context).size.height),
                    color: Colors.yellow
                ),
                child: InkWell(
                  //
                  // onTap: callBack,
                  onTap: (){
                    if(app.numberController.text.trim().isEmpty){
                      Get.snackbar('Number', 'Please Enter Phone Number');
                    }
                    else if(app.numberController.text.trim().toString().length !=11){
                      Get.snackbar('Number', 'Please Enter Valid Phone Number');

                    }
                    else {
                      app.sendRequest(app.numberController.text);
                      // incrementCounter();
                    }
                  },

                  child: Center(child:Text('Send OTP',style: TextStyle(letterSpacing: 2,color: Colors.black,fontSize:MediaQuery.of(context).size.height * 0.02,fontWeight: FontWeight.w500),)),
                ))
          ],
        ),
      ),
    ));
  }
}


example/lib/SuccessScreen.dart
import 'package:example/main.dart';
import 'package:example/otp_selection.dart';
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';
import 'package:get/get.dart';

class SuccessScreen extends StatefulWidget {
  const SuccessScreen({Key? key}) : super(key: key);

  @override
  State<SuccessScreen> createState() => _SuccessScreenState();
}

class _SuccessScreenState extends State<SuccessScreen> {

  @override
  void initState() {
    // TODO: implement initState
    super.initState();
    Future.delayed(Duration(seconds: 3),(){
      Get.offAll(()=>OtpSelection());
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(backgroundColor: Colors.white,
      appBar: AppBar(
        // TRY THIS: Try changing the color here to a specific color (to
        // Colors.amber, perhaps?) and trigger a hot reload to see the AppBar
        // change color while the other colors stay the same.
        backgroundColor: Colors.yellow,
        automaticallyImplyLeading: false,
        // Here we take the value from the MyHomePage object that was created by
        // the App.build method, and use it to set our appbar title.
        title: Text('Callify',style: TextStyle(color: Colors.black,fontWeight: FontWeight.bold),),
        centerTitle: true,
      ),
    body: SafeArea(child:
      Center(child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
        Text('OTP Verified',style: TextStyle(color: Colors.black,fontSize: 20,fontWeight: FontWeight.w700),)
      ],),),),);
  }
}


example/lib/otp_verification.dart
import 'package:awesome_dialog/awesome_dialog.dart';
import 'package:example/SuccessScreen.dart';
import 'package:example/controller.dart';
import 'package:example/otp_selection.dart';
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';
import 'package:flutter_pin_code_fields/flutter_pin_code_fields.dart';
import 'package:get/get.dart';

class OtpVerification extends StatelessWidget {

  var param;
  OtpVerification({@required this.param});

  @override
  Widget build(BuildContext context) {
    return GetBuilder<AppController>(builder: (app)=>Scaffold(
      appBar: AppBar(
        automaticallyImplyLeading: true,
        backgroundColor: Colors.yellow,
        centerTitle: true,
        title: const Text('Callify',style: TextStyle(color: Colors.black),),

      ),
      body: ListView(
        padding: EdgeInsets.all(15.0),
        children: [
          const SizedBox(height: 20,),
           Center(child:param.toString().length == 4 ? Text('Please enter OTP code',style: TextStyle(
             color: Colors.black,fontSize: Get.height * 0.018
           ),) : Text('Please enter last 4 digit of miss call number',style: TextStyle(
               color: Colors.black,fontSize: Get.height * 0.018
           ))),
        SizedBox(height: 15,),
        param.toString().length == 4 ? PinCodeFields(
          textStyle: TextStyle(fontSize: Get.height * 0.035),
          animation: Animations.rotateRight,
          animationDuration: const Duration(milliseconds: 250),
          animationCurve: Curves.bounceInOut,
          switchInAnimationCurve: Curves.bounceIn,
          switchOutAnimationCurve: Curves.bounceOut,
          activeBorderColor: Colors.yellow,
          fieldWidth: 5,
          onComplete: (String value) { app.otpCodeCheck(value); },
        ) :
       Row(children: [
         Expanded(
           child:Text(param.toString().substring(0,param.toString().length - 3),style: TextStyle(fontSize: Get.height * 0.04,
           letterSpacing: 5),)),
         Container(
           width: Get.width * 0.6,child:PinCodeFields(
           textStyle: TextStyle(fontSize: Get.height * 0.035),
           animation: Animations.rotateRight,
           animationDuration: const Duration(milliseconds: 250),
           animationCurve: Curves.bounceInOut,
           switchInAnimationCurve: Curves.bounceIn,
           switchOutAnimationCurve: Curves.bounceOut,
           activeBorderColor: Colors.yellow,
           fieldWidth: 5,
           onComplete: (String value) {
             app.mobileNumberCheck(value);

           },
         ))
       ],),
          const SizedBox(height: 20,),


        Container(

            height: MediaQuery.of(context).size.height * 0.07,


            decoration: BoxDecoration(
                borderRadius: BorderRadius.circular(MediaQuery.of(context).size.height),
                color: Colors.yellow
            ),
            child: InkWell(
              //
              // onTap: callBack,
              onTap: (){
                if(param.toString().length == 4){
                  if(app.userOTP == param.toString()){
                    AwesomeDialog(
                      context: context,
                      animType: AnimType.scale,
                      dialogType: DialogType.success,
                      body: const Center(child: Text(
                        'OTP Verification Successful',
                        style: TextStyle(fontStyle: FontStyle.normal),
                      ),),
                      title: 'This is Ignored',
                      desc:   'This is also Ignored',
                      btnOkOnPress: () {
                        Get.offAll(()=>OtpSelection());
                      },
                    ).show();
                    // Get.to(()=>const SuccessScreen());
                  }
                  else{
                    AwesomeDialog(
                      context: context,
                      animType: AnimType.scale,
                      dialogType: DialogType.error,
                      body: const Center(child: Text(
                        'Invalid OTP - Verification Failed',
                        style: TextStyle(fontStyle: FontStyle.normal),
                      ),),
                      title: 'This is Ignored',
                      desc:   'This is also Ignored',
                      btnOkColor: Colors.red,
                      btnOkOnPress: () {
                        // Get.back();
                      },
                    ).show();

                  }
                }
                else{
                  if(app.userOTP == param.toString().substring(param.toString().length-4)){
                    AwesomeDialog(
                      context: context,
                      animType: AnimType.scale,
                      dialogType:  DialogType.success,
                      body: const Center(child: Text(
                        'OTP Verification Successful',
                        style: TextStyle(fontStyle: FontStyle.normal),
                      ),),
                      title: 'This is Ignored',
                      desc:   'This is also Ignored',

                      btnOkOnPress: () {
                        Get.offAll(()=>OtpSelection());
                      },
                    ).show();
                  }
                  else{
                    AwesomeDialog(
                      context: context,
                      animType: AnimType.scale,
                      dialogType: DialogType.error,
                      body: const Center(child: Text(
                        'Invalid Number - Verification Failed',
                        style: TextStyle(fontStyle: FontStyle.normal),
                      ),),
                      title: 'This is Ignored',
                      desc:   'This is also Ignored',
                      btnOkColor: Colors.red,
                      btnOkOnPress: () {
                       // Get.back();
                      },
                    ).show();


                  }
                }
              },

              child: Center(child:Text('Verify OTP',style: TextStyle(letterSpacing: 2,color: Colors.black,fontSize:MediaQuery.of(context).size.height * 0.02,fontWeight: FontWeight.w500),)),
            ))
      ],),
    ));
  }
}


example/lib/otp_selection.dart
import 'package:example/enums.dart';
import 'package:example/phone_screen.dart';
import 'package:flutter/material.dart';
import 'package:get/get.dart';

class OtpSelection extends StatelessWidget {
  const OtpSelection({super.key});



  @override
  Widget build(BuildContext context) {
    return Scaffold(

      appBar: AppBar(
        backgroundColor: Colors.yellow,
        centerTitle: true,
        title: const Text('Callify',style: TextStyle(color: Colors.black),),
      ),
      body: SafeArea(child: Column(

          children: [
            SizedBox(height: Get.height * 0.2,
            child: Center(child:
            Text('Verification Services',style: TextStyle(decoration: TextDecoration.underline,color: Colors.black,fontSize: Get.height * 0.028,fontWeight: FontWeight.w700),),),),
            Expanded(child: Container(
              width: Get.width,
              decoration: const BoxDecoration(
                borderRadius: BorderRadius.only(topLeft: Radius.circular(40),topRight: Radius.circular(40)),
                color: Colors.yellow
              ),
              child: Column(

                children: [
                  const SizedBox(height: 50,),
                  Row(
                    mainAxisAlignment: MainAxisAlignment.spaceEvenly,

                    children: [
                      InkWell(
                        onTap:(){
                          Get.to(()=>PhoneScreen(),arguments: [VerificationType.missCall]);
                        },
                        child:Container(
                        height: Get.height * 0.16,
                        width: Get.width * 0.35,
                        decoration: BoxDecoration(
                            color: Colors.white,
                            borderRadius: BorderRadius.circular(15),
                            border: Border.all(color: Colors.black,width: 2)
                        ),
                        child: Column(
                          mainAxisAlignment: MainAxisAlignment.center,
                          children: [
                            Image.asset('assets/misscall.png',height: 60,width: 60,),
                            const SizedBox(height: 10,),
                            Text('Miss Call',style: TextStyle(fontSize: Get.height * 0.018),)
                          ],),
                      )),
                      InkWell(
                        onTap:(){
                          Get.to(()=>PhoneScreen(),arguments: [VerificationType.call]);
                        },
                        child:Container(
                        height: Get.height * 0.16,
                        width: Get.width * 0.35,
                        decoration: BoxDecoration(
                            color: Colors.white,
                            borderRadius: BorderRadius.circular(15),
                            border: Border.all(color: Colors.black,width: 2)
                        ),
                        child: Column(
                          mainAxisAlignment: MainAxisAlignment.center,
                          children: [
                            Image.asset('assets/call.png',height: 60,width: 60,),
                            const SizedBox(height: 10,),
                            Text('Call OTP',style: TextStyle(fontSize: Get.height * 0.018))
                          ],),
                      )),
                    ],),
                  const SizedBox(height: 20,),
                  Row(
                    mainAxisAlignment: MainAxisAlignment.spaceEvenly,

                    children: [
                      InkWell(
                        onTap:(){
                          Get.to(()=>PhoneScreen(),arguments: [VerificationType.callify]);
                      },
                        child: Container(
                        height: Get.height * 0.16,
                        width: Get.width * 0.35,
                        decoration: BoxDecoration(
                            color: Colors.white,
                            borderRadius: BorderRadius.circular(15),
                            border: Border.all(color: Colors.black,width: 2)
                        ),
                        child: Column(
                          mainAxisAlignment: MainAxisAlignment.center,
                          children: [
                            Image.asset('assets/callify.png',height: 60,width: 60,),
                            const SizedBox(height: 10,),
                            Text('Callify OTP',style: TextStyle(fontSize: Get.height * 0.018))
                          ],),
                      ),),
                      InkWell(
                        onTap:(){
                          Get.to(()=>PhoneScreen(),arguments: [VerificationType.whatsapp]);
                        },
                        child:Container(
                        height: Get.height * 0.16,
                        width: Get.width * 0.35,
                        decoration: BoxDecoration(
                            color: Colors.white,
                            borderRadius: BorderRadius.circular(15),
                            border: Border.all(color: Colors.black,width: 2)
                        ),
                        child: Column(
                          mainAxisAlignment: MainAxisAlignment.center,
                          children: [
                            Image.asset('assets/whatsapp.png',height: 60,width: 60,),
                            const SizedBox(height: 10,),
                            Text('Whatsapp OTP',style: TextStyle(fontSize: Get.height * 0.018))
                          ],),
                      )),
                    ],),
                  const SizedBox(height: 20,),

                  Row(
                    mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                    children: [
                    InkWell(
                      onTap:(){
                        Get.to(()=>PhoneScreen(),arguments: [VerificationType.sms]);
                      },
                      child: Container(
                        height: Get.height * 0.16,
                        width: Get.width * 0.35,
                        decoration: BoxDecoration(
                            color: Colors.white,
                            borderRadius: BorderRadius.circular(15),
                            border: Border.all(color: Colors.black,width: 2)
                        ),
                        child: Column(
                          mainAxisAlignment: MainAxisAlignment.center,
                          children: [
                            Image.asset('assets/sms.png',height: 60,width: 60,),
                            const SizedBox(height: 10,),
                            Text('SMS OTP',style: TextStyle(fontSize: Get.height * 0.018))
                          ],),
                      ),),
                  ],)
                ],),
            ))])),
    );
  }
}
